server/src/main/java/org/elasticsearch/index/engine/Engine.java

```diff
@@ -175,6 +175,13 @@ public MergeStats getMergeStats() {
     /** Returns how many bytes we are currently moving from heap to disk */
     public abstract long getWritingBytes();
 
+    /**
+     * This method should be called after the translog has been synced.
+     */
+    public void onTranslogSynced() throws IOException {
```

Comment:
I think we should do this differently, rather than sync the translog directly, we can make the translog sync method package private and have all syncs go through the engine. wdyt?